src/modules/hardware-wallet/src/HardwareWallet.ts

```diff
@@ -0,0 +1,80 @@
+import { Instruction, Device, ConnEvent } from './types'
```

Comment:
This module is for all communication to the ledger hardware device.
It defines a generic method for sending so-called APDU messages, which is used by the ledger.
It also defines a method for subscribing to connection events.